Microsoft.Toolkit.Uwp.UI/Extensions/Mouse/Mouse.cs

```diff
@@ -0,0 +1,115 @@
+
+        /// <summary>
+        /// Set the target <see cref="CoreCursorType"/>.
+        /// </summary>
+        /// <param name="element">Object where the selector cursor type should be shown.</param>
+        /// <param name="value">Target cursor type value.</param>
+        public static void SetCursor(FrameworkElement element, CoreCursorType value)
+        {
+            element.SetValue(CursorProperty, value);
+        }
+
+        /// <summary>
+        /// Get the current <see cref="CoreCursorType"/>.
+        /// </summary>
+        /// <param name="element">Object where the selector cursor type should be shown.</param>
+        /// <returns>Cursor type set on target element.</returns>
+        public static CoreCursorType GetCursor(FrameworkElement element)
+        {
+            return (CoreCursorType) element.GetValue(CursorProperty);
+        }
+
+        private static void CursorChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var element = d as FrameworkElement;
+            if (element == null)
+            {
+                throw new NullReferenceException(nameof(element));
+            }
+
+            var value = (CoreCursorType) e.NewValue;
+
+            // lock ensures CoreCursor creation and event handlers attachment/detachment is atomic
+            lock (s_cursorLock)
+            {
+                if (!s_cursors.ContainsKey(value))
+                {
+                    s_cursors[value] = new CoreCursor(value, 1);
+                }
+
+                // make sure event handlers are not attached twice to element
+                element.PointerEntered -= Element_PointerEntered;
+                element.PointerEntered += Element_PointerEntered;
+                element.PointerExited -= Element_PointerExited;
+                element.PointerExited += Element_PointerExited;
+                element.Unloaded -= ElementOnUnloaded;
+                element.Unloaded += ElementOnUnloaded;
+            }
+        }
+
+        private static void Element_PointerEntered(object sender, PointerRoutedEventArgs e)
+        {
+            CoreCursorType cursor = GetCursor((FrameworkElement) sender);
+            Window.Current.CoreWindow.PointerCursor = s_cursors[cursor];
+        }
+
+        private static void Element_PointerExited(object sender, PointerRoutedEventArgs e)
+        {
+            // when exiting change the cursor to the target Mouse.Cursor value of the new element
+            CoreCursor cursor;
+            if (e.OriginalSource is FrameworkElement newElement)
+            {
+                cursor = s_cursors[GetCursor(newElement)];
+            }
+            else
+            {
+                cursor = s_defaultCursor;
+            }
+
+            Window.Current.CoreWindow.PointerCursor = cursor;
+        }
+
+        private static void ElementOnUnloaded(object sender, RoutedEventArgs routedEventArgs)
+        {
```

Comment:
Unsubscribe from pointer events

Reply:
Not sure how this could be solved or whether there is need for unsubscribing those events? It does not memory leak if that is what are you worried about.
One solution could be changing the default value of this property to **DependencyProperty.UnsetValue** and unsubscribing all events when the property is changed manually to this value?

Reply:
I'm pretty sure this will cause the element to remain and not get garbage collected

Reply:
Event handlers in static class are at the end of object graph. I've tested this and object with Mouse.Cursor set are garbage collected just fine.